### PR TITLE
fix: too many open files installing istio addon

### PR DESCRIPTION
**Pull request by carlossg**
Each file inside the tarball is kept open while tarball is extracted

Signed-off-by: Carlos Sanchez <carlos@apache.org>

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/util/downloads.go [hunk 1]
```diff
@@ -252,23 +252,7 @@ func UnTargz(tarball, target string, onlyFiles []string) error {
...
-				return err
-			}
-			continue
-		}
-
-		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
-		if err != nil {
-			return err
-		}
-		defer file.Close()
```
Comment (carlossg) [inline pkg/util/downloads.go]: this defer was not called until the function ended, so each file inside the for loop was open and not closed

**Comment by hferentschik** [comment]
Do we have an issue for that? IMO it would be nice to capture these things in issues. I imagine that there was an error or panic occurring. For reference, it would be nice to have this information captured in an issue together with the fix.

Also, could one write a test verifying that this issue is fixed?

**Comment by carlossg** [comment]
There's no panic, just the `too many open files` error
What's the point of having issues for PRs? just makes the information duplicated and harder to find and maintain

**Comment by hferentschik** [comment]
> There's no panic, just the too many open files error

Which would have been useful context. Taking this "too many open files" as search string on the issues tab gets me https://github.com/jenkins-x/jx/issues/582 where I see the context out of which this whole thing arose. Note, that this search out of the box did not return this PR.

GitHub does in many regards abstract away the difference between pull requests and issues, but conceptually they are different things and even on a technical / UI level this difference is noticeable. 

> What's the point of having issues for PRs? just makes the information duplicated and harder to find and maintain

IMO the contrary is true. If we were to use an external issue tracker like JIRA, would you still say that creating an issue would not be necessary?

**Comment by carlossg** [comment]
> If we were to use an external issue tracker like JIRA, would you still say that creating an issue would not be necessary?

in that case it would, because people would search in jira for issues. Here they can search in both issues and PRs at once

**Comment by hferentschik** [comment]
> in that case it would, because people would search in jira for issues. Here they can search in both issues and PRs at once

You can, but it is not necessarily obvious

**Pull request merged**